### ISSUE TITLE
`c1_deep_copy` Non-SCF Wfn Costs Orbitals

### ISSUE THREAD

**Issue by JonathonMisiewicz** [psi4/psi4#2777]
```
molecule {
O
H 1 1.0 
H 1 1.0 2 104.5
}

wfn = energy("mp2/sto-6g", return_wfn=True)[1]
new_wfn = wfn.c1_deep_copy(wfn.basisset())
new_wfn.Ca()
```

Using `c1_deep_copy` on a non-HF wfn means the orbitals don't copy: that's handled by the constructor.

This is causing problems in Francesco's group.

**Comment by loriab** [comment]
Looks like the plain `deep_copy` does the same.

**Comment by loriab** [comment]
Are you describing that `c1_deep_copy` on R/U/ROHF wfns are fine b/c https://github.com/psi4/psi4/blob/1ecf33397370ebaa3b2ec14dd276c22519c66ed4/psi4/src/psi4/libscf_solver/rhf.cc#L999-L1002 etc. in the derived classes copy over the orbitals; whereas, `c1_deep_copy` on say, CC, is lacking because those derived wfn classes haven't extended the method?

This makes sense historically since `c1_deep_copy`'s purpose was simply transitioning a symmetry-using SCF wfn into a c1-only post-SCF module. But I can see its deficiencies beyond that.

**Comment by JonathonMisiewicz** [comment]
Yes, the HF subclasses are fine. More general wavefunctions are not. We stumbled across it by trying to take a CASSCF wavefunction, convert it to c1, and feed that the de-symmetrized orbitals into an orbital localizer. This is a different use case than passing a reference!

I would think that the `hf.cc` class can figure out what it needs to do, based on checking `same_a_b_orbs` and `same_a_b_dens`. Other subclasses should only need to extend if it needs anything more sophisticated than that.
